ledger/simulation/testing/utils.go

```diff
@@ -67,17 +69,11 @@ func (info TxnInfo) NewTxn(txn txntest.Txn) txntest.Txn {
 	return txn
 }
 
-// InnerTxn sets network- and parent-specific values to the given inner transaction. This is only
-// useful for creating an expected inner transaction to compare against.
-func (info TxnInfo) InnerTxn(parent transactions.SignedTxn, inner txntest.Txn) txntest.Txn {
-	inner.FirstValid = parent.Txn.FirstValid
-	inner.LastValid = parent.Txn.LastValid
-	inner.FillDefaults(info.CurrentProtocolParams())
-	return inner
-}
-
-// Environment contains the ledger and testing environment for transaction simulations
+// Environment contains the ledger and testing environment for transaction simulations. It also
+// provides convenience methods to execute transactions against the ledger prior to simulation. This
+// allows you to create specific a ledger state before running a simulation.
```

Comment:
Did you write all these Environment methods this week?  or are these from somewhere?

Reply:
They're unique to the PR, but I wrote them weeks ago